include/pybind11/detail/internals.h

```diff
@@ -280,21 +281,112 @@ inline internals **&get_internals_pp() {
     return internals_pp;
 }
 
+#if PY_VERSION_HEX >= 0x03030000
+inline bool raise_err(PyObject *exc_type, const char *msg) {
+    bool err_occurred = PyErr_Occurred() != nullptr;
+    if (err_occurred) {
+        raise_from(exc_type, msg);
+    } else {
+        PyErr_SetString(exc_type, msg);
+    }
+    return err_occurred;
+};
+
+// forward decl
+inline void translate_exception(std::exception_ptr);
+
+template <
+    class T,
+    enable_if_t<std::is_same<std::nested_exception, remove_cv_t<remove_reference_t<T>>>::value,
+                int> = 0>
+bool handle_nested_exception(const T &exc, const std::exception_ptr &p) {
+    std::exception_ptr nested = nullptr;
+    nested = exc.nested_ptr();
+    if (nested != nullptr && nested != p) {
+        translate_exception(nested);
+        return true;
+    }
+    return false;
+}
+
+template <
+    class T,
+    enable_if_t<!std::is_same<std::nested_exception, remove_cv_t<remove_reference_t<T>>>::value,
+                int> = 0>
+bool handle_nested_exception(const T &exc, const std::exception_ptr &p) {
+    if (auto *nep = dynamic_cast<const std::nested_exception *>(std::addressof(exc))) {
+        return handle_nested_exception(*nep, p);
+    }
+    return false;
+}
+#else
+template <class T>
+void handle_nested_exception(T, std::exception_ptr) {
```

Comment:
This is `void` while the >=3.3 version returns a `bool`. Is that intentional? A terse comment to explain would be good.